packages/cli/src/commands/generate.ts

```diff
@@ -0,0 +1,150 @@
+    }),
+  }
+
+  static examples = [
+    `# We can just generate a single file type, for example a query:
+blitz generate query task
+    `,
+    `# Or, we can generate a full set of models, mutations, pages, etc.
+blitz generate resource task
+    `,
+    `# If we have multiple entries in /app, we must specify where the files belong
+blitz generate resource task -p=taskManager`,
+  ]
+
+  async promptForTargetDirectory(paths: string[]): Promise<string> {
+    return enquirer
+      .prompt<{directory: string}>({
+        name: 'directory',
+        type: 'select',
+        message: 'Please select a target directory:',
+        choices: paths,
+      })
+      .then((resp) => resp.directory)
+  }
+
+  async genericConfirmPrompt(message: string): Promise<boolean> {
+    return enquirer
+      .prompt<{continue: string}>({
+        name: 'continue',
+        type: 'select',
+        message: message,
+        choices: ['Yes', 'No'],
+      })
+      .then((resp) => resp.continue === 'Yes')
+  }
+
+  async run() {
+    const {args, flags}: {args: Args; flags: Flags} = this.parse(Generate)
+    debug('args: ', args)
+    debug('flags: ', flags)
+
+    console.log(args, flags)
+
+    const isInRoot = fs.existsSync(path.resolve('blitz.config.js'))
+
+    if (!isInRoot) {
+      this.error(chalk.red('No blitz.config.js found. `generate` must be run from the root of the project.'))
+    }
+
+    // we can only assume the output directory if /app has a single child directory,
+    // otherwise we should prompt the user
+    let appPaths = fs.readdirSync(path.resolve('app'))
+    let outputTarget: string
+    if (appPaths.length === 1) {
+      // if there's only one path, we can ignore the path flag safely
+      const appPath = appPaths[0]
+      if (flags.path && flags.path !== appPath) {
+        // if the path provided isn't the only path, warn the user
+        this.log(
+          chalk.yellow(
+            `Warning: supplied path '${flags.path}' doesn't exist in this project. The only found path was '${appPath}.'`,
+          ),
+          {pad: true},
+        )
+        const shouldContinue = await this.genericConfirmPrompt(
+          `Would you like to place files in '/app/${appPath}' instead?`,
+        )
+        if (!shouldContinue) {
+          this.exit(0)
+        }
+      }
+      outputTarget = appPaths[0]
+    } else {
+      // otherwise, validate the provided path, prompting the user if it's absent or invalid
+      if (!flags.path) {
+        this.log(
+          chalk.yellow(
+            `No path flag (--path, -p) was detected, but multiple roots under /app were present (${appPaths.join()})`,
+          ),
+        )
+        outputTarget = await this.promptForTargetDirectory(appPaths)
+      } else if (!appPaths.includes(flags.path)) {
+        this.log(chalk.yellow(`Given path '${flags.path}' doesn't exist in /app. Please select a valid path`))
```

Comment:
All this should ideally use our `log.ts` stuff.

Reply:
yeah this was addressed in the PR description and in @flybayer's comment